File: lib/email_inquire/validator/common_provider.rb
# frozen_string_literal: true

require "damerau-levenshtein"
require "email_inquire/validator/base"

module EmailInquire
  module Validator
    class CommonProvider < Base

      DOMAINS = load_data("common_providers").freeze

      def validate
        response.valid! if DOMAINS.include?(domain)
      end

    end
  end
end

File: implementations/dotnet/EmailInquire/Validators/CommonProvider.cs
using System.Linq;

namespace EmailInquire.Validators
{
    /// <summary>
    ///     Validation over common provider list
    /// </summary>
    public class CommonProvider : ValidatorBase<CommonProvider>
    {
        public override Response Validate()
        {
            return CommonDomains?.Contains(Domain) ?? false ? Response.Valid(Email) : Response.Undefined;
        }
    }
}